doc/source/f2py/asterisk2.f90

```diff
@@ -0,0 +1,6 @@
+subroutine foo2(s, n)
+  character(len=*), intent(out) :: s
+  integer, intent(in) :: n
+  !f2py character(f2py_len=n), depend(n) :: s
+  s = "123456789A123456789B"(1:n)
```

Comment:
In the same vein as the previous discussion. It seems like a natural assumption is for `f2py` to assume the following (which works right now):

```fortran
subroutine foo2(s, n)
  character(len=n), intent(out) :: s
  integer, intent(in) :: n
  s = "123456789A123456789B"(1:n)
end subroutine foo2
```

(note the change from * to n)

Reply:
Sure, the given example gives a well-defined behavior and is preferred. However, the purpose here is to test the `character(len=*)` case, not `character(len=n)` :)